pkg/build/buildutil/util.go

```diff
@@ -257,8 +272,18 @@ func GetBuildSystemConfigMapName(build *buildv1.Build) string {
 // label in a pod. If the length of the string parameter exceeds
 // the maximum label length, the value will be truncated.
 func LabelValue(name string) string {
-	if len(name) <= validation.DNS1123LabelMaxLength {
-		return name
+	end := len(name)
+	newName := name
+	// first, try to truncate from the end to find a valid
+	// label
+	for end > 0 {
+		errStrs := validation.IsDNS1123Label(newName)
+		if len(errStrs) == 0 {
+			return newName
+		}
+		end--
+		newName = newName[:end]
 	}
-	return name[:validation.DNS1123LabelMaxLength]
+	klog.Warningf("In creating the value of the build label in the build pod, several attempts at manipulating %s to meet k8s label name requirements failed", name)
```

Comment:
Will we ever log this - would something like an all-symbols label value fail?

Reply:
yes which is why I left it ... agree the possibility is remote, but seems more positive than negative to me